CONTRIBUTING.md

```diff
@@ -1,100 +1,90 @@
 
-    * Make sure that the pull request message contains all important 
-      information from the commit messages including which issues are
-      fixed and closed. If a pull request contains one commit only, then
-      repeating the commit message is preferred, which is done automatically
-      by GitHub when it creates the pull request.
+Each RFC stage builds on community engagement and ensures that proposed features or changes align with project goals.
+Maintainers will assign the right stage of Issues and PR's through GitHub labels.
 
-    * Rebase your local feature branch against main (`git fetch origin`,
-      then `git rebase origin/main`) to make sure you're
-      submitting your changes on top of the newest version of our code.
+## Contributing code
 
-    * When finalizing your PR (i.e., it has been approved for merging),
-      aim for the fewest number of commits that
-      make sense. That is, squash any "fix up" commits into the commit they
-      fix rather than keep them separate. Each commit should represent a
-      clean, logical change and include a descriptive commit message.
+### Contributor License Agreement (CLA)
 
-    * Push your commit to your upstream feature branch: `git push -u <yourfork> my-feature-branch`
+Before we accept any code contributions, pgai contributors need to sign the [Contributor License Agreement (CLA)](https://cla-assistant.io/timescale/pgai). By signing a CLA, we ensure that the community is free and confident in its ability to use your contributions.
 
-* Create and manage pull request:
+### Creating a Pull Request:
 
-    * [Create a pull request using GitHub](https://help.github.com/articles/creating-a-pull-request).
-      If you know a core developer well suited to reviewing your pull
-      request, either mention them (preferably by GitHub name) in the PR's
-      body or [assign them as a reviewer](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/).
+Once your RFC is accepted (and in the case you didn't open a PR yet), or for smaller changes that do not require the RFC process, you are ready to open a PR.
+Please strictly follow the guidelines below:
 
-    * Address feedback by amending your commit(s). If your change contains
-      multiple commits, address each piece of feedback by amending that
-      commit to which the particular feedback is aimed.
+1. Fork the repository and create a new branch for your work.
+2. Write clean, well-documented, and tested code.
+3. Ensure that your code adheres to the project's style and guidelines. 
+4. Include relevant tests and documentation updates.
+5. Don't mix bug fixes with unrelated feature enhancements or stylistical changes.
+6. The project uses [conventional commits][conventional-commits]. It's enforced by CI, you won't be able to merge PRs if your commits do not comply. This helps us automate the release process, changelog generation, etc. For example:
```

Comment:
Please add the following to the bottom of the page:

[conventional-commits]: https://www.conventionalcommits.org/en/v1.0.0/

Reply:
You mean at the bottom of the Create a Pull Request section?

Reply:
link fixed 👍